iaa: Add e2e tests

ozhuraki opened the pull request:
Closes: #949

mythi commented on demo/dsa-accel-config-demo/Dockerfile:
```diff
@@ -4,7 +4,7 @@ RUN dnf install -y \
     libuuid-devel json-c-devel
 
-ARG ACCEL_CONFIG_VERSION=v3.4.6.3
+ARG ACCEL_CONFIG_VERSION=v3.4.6.4
```
this can go to a PR of it's own (with the same version updated to `intel-idxd-config-initcontainer`)

bart0sh replied:
@mythi done: #1038

ozhuraki commented:
Changes:
- Dropped accel-config version update from idxd-config-initcontainer, dsa-accel-config-demo
- Added IAA demo runner improvements by @bart0sh 
- Renamed dsa-accel-config-demo image to accel-config-demo as discussed

ozhuraki commented on cmd/iaa_plugin/iaa_plugin.go:
```diff
@@ -28,11 +28,11 @@ const (
 	namespace = "iaa.intel.com"
 	// SysFS directory.
-	sysfsDir = "/sys/bus/iax/devices"
+	sysfsDir = "/sys/bus/dsa/devices"
```
This has changed since we initially validated the plugin. Both iax and dsa are exposed by the current idxd kernel driver under the "/sys/bus/dsa".

bart0sh commented on cmd/iaa_plugin/iaa_plugin.go:
```diff
@@ -28,11 +28,11 @@ const (
 	// Device plugin settings.
 	namespace = "iaa.intel.com"
 	// SysFS directory.
-	sysfsDir = "/sys/bus/iax/devices"
```
@ozhuraki do we really need this? Can you point me out where sysfsDir is used?

bart0sh replied:
this can also be done as a separate PR.

mythi commented on .github/workflows/ci.yaml:
```diff
@@ -115,7 +115,7 @@ jobs:
           # Demo images
           - crypto-perf
-          - dsa-accel-config-demo
+          - accel-config-demo
```
renaming should go to a PR of its own.

mythi commented on deployments/iaa_plugin/overlays/iaa_initcontainer/iaa_initcontainer.yaml:
```diff
@@ -13,7 +13,7 @@ spec:
                 fieldPath: spec.nodeName
           - name: IDXD_DEVICE_TYPE
-            value: "iaa"
+            value: "iax"
```
It looks this is just a device identifier that needs to be in sync with the json conf file name. Is there a reason for renaming?

ozhuraki replied:
This way it's in sync with json and idxd-init.sh

bart0sh commented on deployments/iaa_plugin/base/intel-iaa-plugin.yaml:
```diff
@@ -33,7 +33,7 @@ spec:
           readOnly: true
         - name: sysfs
-          mountPath: /sys/bus/iax
+          mountPath: /sys/bus/dsa
```
should it be narrowed to /sys/bus/dsa/devices?

bart0sh commented on cmd/dsa_plugin/dsa_plugin.go:
```diff
@@ -46,7 +44,7 @@ func main() {
 	}
 
-	plugin := idxd.NewDevicePlugin(sysfsDir, statePattern, devDir, sharedDevNum)
+	plugin := idxd.NewDevicePlugin("", statePattern, devDir, sharedDevNum)
```
What's the point to pass this parameter if it's not used?

ozhuraki commented:
Changes:
- Dropped renaming, this will be done as a separate PR
- Dropped sysfs elimination from pkg/idxd, will go into separate PR too

mythi commented:
> * Dropped sysfs elimination from pkg/idxd, will go into separate PR too

@ozhuraki can you rebase, the IAA part of the `sysfsPath` removal was left in this PR. thanks!

mythi reviewed (approved):
thanks!

bart0sh reviewed (approved):
/lgtm
